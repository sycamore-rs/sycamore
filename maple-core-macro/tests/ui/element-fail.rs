use maple_core::prelude::*;

fn compile_fail() {
    template! { p.my-class#id };

    template! { button(disabled) };
    template! { button(on:click) };
    template! { button(unknown:directive="123") };

    template! { button(a.b.c="123") };
}

fn main() {}
