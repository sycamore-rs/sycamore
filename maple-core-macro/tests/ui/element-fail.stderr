error: unexpected token
 --> $DIR/element-fail.rs:4:18
  |
4 |     template! { p.my-class#id };
  |                  ^

error: expected `=`
 --> $DIR/element-fail.rs:6:23
  |
6 |     template! { button(disabled) };
  |                       ^^^^^^^^^^

error: expected `=`
 --> $DIR/element-fail.rs:7:23
  |
7 |     template! { button(on:click) };
  |                       ^^^^^^^^^^

error: unknown directive `unknown`
 --> $DIR/element-fail.rs:8:24
  |
8 |     template! { button(unknown:directive="123") };
  |                        ^^^^^^^

error: expected `=`
  --> $DIR/element-fail.rs:10:25
   |
10 |     template! { button(a.b.c="123") };
   |                         ^
